inference-engine/tools/compile_tool/main.cpp

```diff
@@ -216,6 +239,31 @@ static InferenceEngine::Precision getOutputPrecision(const std::string &value) {
     return getPrecision(value, supported_precisions, "for output layer");
 }
 
+static InferenceEngine::Layout getLayout(const std::string &value) {
+    static const supported_layouts_t supported_layouts = {
+            { "NCHW", InferenceEngine::Layout::NCHW },
+            { "NHWC", InferenceEngine::Layout::NHWC },
+            { "NC", InferenceEngine::Layout::NC },
+            { "C", InferenceEngine::Layout::C }
+    };
+    return getLayout(value, supported_layouts);
+}
+
+static bool isMatchLayoutToDims(const InferenceEngine::Layout& layout, const size_t dimension) {
+    static const matchLayoutToDims_t matchLayoutToDims = {
+            { std::hash<InferenceEngine::Layout>{}(InferenceEngine::Layout::NCHW), 4 },
+            { std::hash<InferenceEngine::Layout>{}(InferenceEngine::Layout::NHWC), 4 },
+            { std::hash<InferenceEngine::Layout>{}(InferenceEngine::Layout::NC), 2 },
+            { std::hash<InferenceEngine::Layout>{}(InferenceEngine::Layout::C), 1 }};
```

Comment:
`static_cast<size_t>` instead of `std::hash<InferenceEngine::Layout>{}`

Reply:
Fixed.